tests/lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,422 @@
+/**
+ * @fileoverview Prefers object spread property over Object.assign
+ * @author Sharmila Jesupaul
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+const rule = require("../../../lib/rules/prefer-object-spread");
+
+const RuleTester = require("../../../lib/testers/rule-tester");
+
+const parserOptions = {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+        experimentalObjectRestSpread: true
+    }
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+
+ruleTester.run("prefer-object-spread", rule, {
+    valid: [
+        "Object.assign()",
+        "let a = Object.assign(a, b)",
+        "Object.assign(a, b)",
+        "let a = Object.assign(b, { c: 1 })",
+        "let a = Object.assign({}, ...b)",
+        "const bar = { ...foo }",
+        "Object.assign(...foo)",
+        "Object.assign(foo, { bar: baz })"
+    ],
+
+    invalid: [
+        {
+            code: "Object.assign({}, foo)",
+            output: "({...foo})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar' })",
+            output: "({foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, baz, { foo: 'bar' })",
+            output: "({...baz, foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar', baz: 'foo' })",
+            output: "({foo: 'bar', baz: 'foo'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({ foo: 'bar' }, baz)",
+            output: "({foo: 'bar', ...baz})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Many args
+        {
+            code: "Object.assign({ foo: 'bar' }, cats, dogs, trees, birds)",
+            output: "({foo: 'bar', ...cats, ...dogs, ...trees, ...birds})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // TODO: Handle nested Object.assign calls
```

Comment:
Should this comment be removed? (Although, now I see some of the test cases below, seems there might be more to do here?)